src/utils/daos.c

```diff
@@ -881,6 +907,11 @@ common_op_parse_hdlr(int argc, char *argv[], struct cmd_args_s *ap)
 
 	cmd_args_print(ap);
 
+	if (posix_mode_set && ap->type != DAOS_PROP_CO_LAYOUT_POSIX) {
+		fprintf(stderr, "--mode is valid only for a POSIX container\n");
+		D_GOTO(out_free, rc = RC_NO_HELP);
+	}
+
```

Comment:
This might need updating for the go implementation.

Reply:
im not sure what you mean. can you suggest the change?

Reply:
I was thinking that this file isn't used any more other than for the daos_old command, and the change here will need to be ported to the new go code.

Reply:
OK i didn't realize we replaced the entire daos.c main function. 
i guess i will also have missed the mode option in the go frontend..

Reply:
it seems @mjmac had already copied this from my PR and added it to the go container frontend.
so i think we are good as it's only under type posix:
https://github.com/daos-stack/daos/commit/c66eef2775df7e322213906f2ae7ab6f29c3d16d#diff-c8c509b1b2b0c108d3c02f9e16de02a07c81ad5a920d5c72617198a5bc7b548dR185

Reply:
I think so, I've got the same issue with https://github.com/daos-stack/daos/pull/5969

Given this is disabled by default it shouldn't block this PR however but you do want to test this.

Reply:
im going to add some NLT tests for this in a different PR

Reply:
Is this file dead code?

Reply:
It's `daos_old` now:
https://github.com/daos-stack/daos/blob/c66eef2775df7e322213906f2ae7ab6f29c3d16d/src/utils/SConscript#L45-L49
Presumably to be removed at some later date?